readthedocs/doc_builder/environments.py

```diff
@@ -792,7 +792,7 @@ def __init__(self, *args, **kwargs):
         self.container = None
         self.container_name = slugify(
             'build-{build}-project-{project_id}-{project_name}'.format(
-                build=self.build.get('id'),
+                build=self.build.get('id') if self.build else '',
```

Comment:
The name may collapse if two tasks are triggered in the same lapse of time. Not sure what happens there.

Reply:
Just to answer my own question, we do a check for this in the __enter__ event.